src/goto-programs/goto_convert.cpp

```diff
@@ -1142,7 +1142,11 @@ void goto_convertt::convert_assert(
   t->guard.swap(cond);
   t->source_location=code.source_location();
   t->source_location.set(ID_property, ID_assertion);
-  t->source_location.set("user-provided", true);
+  irep_idt property_class=t->source_location.get_property_class();
+  // if it's not our instrumentation for runtime exceptions, then it must be
+  // provided by the user
+  if(id2string(property_class)!="runtime-exception")
```

Comment:
Instead of using `id2string`, add these to `irep_ids.def` (all common quoted strings, e.g. runtime-exception, user-provided, ...)